clang/test/CodeGenSYCL/sycl-device-alias.cpp

```diff
@@ -0,0 +1,25 @@
+// RUN: %clang_cc1 -fsycl -fsycl-is-device -triple spir64-unknown-unknown-sycldevice -disable-llvm-passes -emit-llvm %s -o - | FileCheck %s
+// Test that aliasing does not force an unused function to be emitted
+
+// CHECK-NOT: define spir_func void @unused_func()
+extern "C" void unused_func() {}
```

Comment:
Seems like `alias` attribute can be also applied to a global variable. Could you please add corresponding tests?

Reply:
Thanks, I did this.  (I didn't change the title to avoid needing to force-push, but I will do that when there are no further code-review comments.)